core/dbt/cli/flags.py

```diff
@@ -112,7 +112,10 @@ def assign_params(ctx, params_assigned_from_default, deprecated_env_vars):
                         # param already set via its deprecated but still respected env var
                         continue
 
-                    if param_name not in EXPECTED_DUPLICATE_PARAMS:
+                    if (
+                        param_name not in EXPECTED_DUPLICATE_PARAMS
+                        and ctx.get_parameter_source(param_name) != "kwargs"
```

Comment:
Just checking my understanding - will the parameter value from `kwargs` here will clobber the parameter value from the `args` list?